WordPress/gutenberg
# Remove base control negative help text margin

## Description
In https://github.com/WordPress/gutenberg/pull/29012#issuecomment-785308585, @shaunandrews noticed that help text for the BaseControl component looks uncomfortably close to the input itself. In the navigation editor it looks like this:
<img width="244" alt="Screenshot 2021-03-04 at 3 28 50 pm" src="https://user-images.githubusercontent.com/677833/109926828-600e0100-7cfe-11eb-9ffd-ab32bb24b72e.png">

There's some negative margin causing the help text not to social distance from the input.

In the post editor, it generally looks like this:
<img width="278" alt="Screenshot 2021-03-04 at 3 30 23 pm" src="https://user-images.githubusercontent.com/677833/109926992-8fbd0900-7cfe-11eb-869b-81287b79692c.png">

Negative margin isn't applied in the sidebar because of this non-BEM CSS that styles every sidebar paragraph:
<img width="229" alt="Screenshot 2021-03-04 at 3 25 34 pm" src="https://user-images.githubusercontent.com/677833/109927106-ae230480-7cfe-11eb-90dc-2c30e0716c73.png">

Seems like an inconsistency. This PR removes negative margin from BaseControl so that the navigation editor looks consistent.

## How has this been tested?
1. Open up the navigation editor
2. Open up the popover for adding a new menu
3. Notice things look nicer.

## Screenshots 
<img width="272" alt="Screenshot 2021-03-04 at 3 34 51 pm" src="https://user-images.githubusercontent.com/677833/109927440-33a6b480-7cff-11eb-942c-3a0fe45d0068.png">

## Types of changes
Minor breaking CSS change.





## Checklist:
